### PR TITLE
Fixes for Flagging UI Component Dependencies

### PR DESCRIPTION
**Pull request by gbashour**
- Removed imports not being used in index.js
- Added socket.io and socket.io-client imports in server and client package.json

### CODE REVIEW

**Comment by gbashour** [comment]
Tested Locally👍

File: client/src/components/AdminFlagNotification.js [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+
+const AdminFlagNotification = () => {
+  const [flagAdded, setFlagAdded] = useState(false);
+  const [flagUpdated, setFlagUpdated] = useState(false);
+  const [isAdmin, setIsAdmin] = useState(false);
+  const [flagId, setFlagId] = useState("");
+  const [student, setStudent] = useState('');
+  const [studentId, setStudentId] = useState('');
+
+  const url = 'http://localhost:4000' + '/flag';
```
Comment (abrarbarbhuyia) [inline client/src/components/AdminFlagNotification.js]: This needs to be process.env.REACT_APP_SERVER_URL then + '/flag'

File: client/src/components/AdminFlagNotification.js [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+        setStudent(studentData);
+        setStudentId(studentId); // Store the studentId
+        console.log(studentId);
+      })
+      .catch(studentError => {
+        console.error(studentError);
+      });
+    }
+
+    const socket = io('http://localhost' + ':4001'); // Change the URL to match the Socket.IO server URL
```
Comment (abrarbarbhuyia) [inline client/src/components/AdminFlagNotification.js]: Same as this one env var + ':4001'

File: client/src/components/FlagNotification.js [hunk 3]
```diff
@@ -37,9 +32,10 @@ const FlagNotification = () => {
       });
     }
 
-    const socket = io(process.env.REACT_APP_SERVER_URL + ':4001'); // Change the URL to match the Socket.IO server URL
+    const socket = io('http://localhost' + ':4001'); // Change the URL to match the Socket.IO server URL
```
Comment (abrarbarbhuyia) [inline client/src/components/FlagNotification.js]: Same here

File: server/routes/flag.js [hunk 4]
```diff
@@ -10,7 +10,7 @@ var connectedId;
 
 const io = require('socket.io')(PORT, {
     cors: {
-        origin: ['https://sentinel-frontend.vercel.app'] // Client URL -- http://localhost:3000 for testing
+        origin: ['http://localhost:3000'] // Client URL -- http://localhost:3000 for testing
```
Comment (abrarbarbhuyia) [inline server/routes/flag.js]: Another one

**Review by abrarbarbhuyia — CHANGES_REQUESTED** [review]
LGTM, just some URLs need fixing

File: server/routes/flag.js [hunk 5]
```diff
@@ -10,7 +10,7 @@ var connectedId;
 
 const io = require('socket.io')(PORT, {
     cors: {
-        origin: ['http://localhost:3000'] // Client URL -- http://localhost:3000 for testing
```
Comment (abrarbarbhuyia) [inline server/routes/flag.js]: Might need to add FRONTEND_URL="" in .env for server.
Then call it with process.env.FRONTEND_URL here

**Review by abrarbarbhuyia — APPROVED** [review]

File: server/routes/flag.js (hunk 5 above)
Comment (abrarbarbhuyia) [reply server/routes/flag.js]: OMG Ignore me, sorry.

Can you change this back to
const corsOptions = {
  origin: ['http://localhost:3000', 'https://sentinel-frontend.vercel.app'],
};

T_T

**Review by abrarbarbhuyia — APPROVED** [review]

**Pull request merged**